src/main/java/org/elasticsearch/index/mapper/geo/GeoPointFieldMapper.java

```diff
@@ -254,25 +250,27 @@ public GeoPointFieldMapper build(BuilderContext context) {
                     }
                     iterator.remove();
                 } else if (fieldName.equals("validate")) {
-                    builder.validateLat = XContentMapValues.nodeBooleanValue(fieldNode);
-                    builder.validateLon = XContentMapValues.nodeBooleanValue(fieldNode);
+                    builder.validate = XContentMapValues.nodeBooleanValue(fieldNode);
                     iterator.remove();
                 } else if (fieldName.equals("validate_lon")) {
-                    builder.validateLon = XContentMapValues.nodeBooleanValue(fieldNode);
-                    iterator.remove();
+                    if (parserContext.indexVersionCreated().onOrAfter(Version.V_1_6_0)) {
+                        throw new ElasticsearchParseException("'validate_lon' is deprecated and will be removed in the next release");
```

Comment:
The message is a little weird.  I don't think "next release" should be mentioned.